### ISSUE TITLE
As a data editor, I want to enter months for periodical publication dates and non-numeric volumes when entering multivolume and periodical details.

### ISSUE THREAD

**Issue by rlskoeser** [Princeton-CDH/mep-django#518]
## testing notes
- test that you can add a non-numeric value for issue (e.g. that 19-21 issue example)
- confirm that you see help text for the season field
- test that you can add a partial date: try no date, year only, year + month

## dev notes
- [x] convert edition issue number from numeric to string 
- [x] add help text to season to indicate it should include season or month spelled out if it should be included in the issue citation: (wording suggestion: "Spell out month or season if part of numbering")
- [x] convert edition year to a partial date field
- [x] add a migration to convert existing years to partial dates with year precision
- [x] configure partial date editing in admin

**Comment by rlskoeser** [comment]
@jkotin @clmahoney please comment with any other changes you're aware of now based on Cate's testing.

I propose converting the year field to a partial date: this would accommodate the months Cate found for one journal and would also be used for sorting.

**Comment by rlskoeser** [comment]
@jkotin I've done a test implementation of the volumes/issues list based on Gissoo's design for the bibliography page to make sure I can generate the display and to check on the data model changes for Edition records.

Here are the changes I propose:
- convert issue **number** from numeric to string so we can handle non-numeric issues like "19-20"
- add help text to **season** to indicate it should include season or month spelled out if it should be included in the issue citation (wording suggestions appreciated)
- convert **year** to a partial date field so that it can include month (and even day) when a more specific date is known; only the year will be shown when the issues/volumes are listed, but month/day will be used for sorting

Question:
- Will **volume** always be numeric?

Please review these proposed changes and confirm your agreement or let me know any adjustments so that we can proceed with implementing them.

**Comment by jkotin** [comment]
@rlskoeser This all sounds good. I'm tagging @clmahoney so she can confirm -- she's done the most work on the volumes/issues. For help text for season, how about "Spell out month or season if part of numbering"?  That's not very elegant.  I'm happy for alternatives.

**Comment by rlskoeser** [comment]
@clmahoney please review and let me know if these changes sound like what we need based on the volume/issue data work you've done.

One thing that occurred to me: entering month in both the partial date and as text in the "season" field is redundant, and we have an existing template tag that will let me easily display month and year if month is known, or just year if month is not known. The one problem with that is if we need more control over the display: will you ever know the month but not want it displayed? (e.g. if there is also a season)

**Comment by clmahoney** [comment]
@rlskoeser yes, volume is always numeric. 
Re: season v. month. Usually it's one or the other; rarely both. I think it would be most helpful to see the month, since that can narrow down the most quickly.

**Comment by clmahoney** [comment]
- YES, I can add non-numeric values. Tried with that two issue span and it works! See note: 
![Screen Shot 2020-03-21 at 8 24 39 AM](https://user-images.githubusercontent.com/16673934/77226311-b9ad0800-6b4d-11ea-8b31-f7f79c294881.png)
- YES, I see the season field help text: 
![Screen Shot 2020-03-21 at 8 25 12 AM](https://user-images.githubusercontent.com/16673934/77226314-c5003380-6b4d-11ea-8763-a08fa76d6342.png)
- YES, partial dates work too! I tried year only and only month and date: 
![Screen Shot 2020-03-21 at 8 26 02 AM](https://user-images.githubusercontent.com/16673934/77226326-d8130380-6b4d-11ea-8108-c4ab85d52cfd.png). 

Looks great!
